test/collection.jl

```diff
@@ -0,0 +1,45 @@
+using CCDReduction: match_extension
+
+@testset "fitscollection" begin
+    # setting initial data
+    dir = joinpath(@__DIR__, "data")
+
+    # the two fits file used for testing have identical header keys
+    df = fitscollection(dir)
+    relevant_keys = filter(i -> i ∉ ("", "HISTORY"), keys(read_header(M6707HH[1])))
+    relevant_values_M6707HH = map(relevant_keys) do key
+        read_header(M6707HH[1])[key]
+    end
+    relevant_values_M35070V = map(relevant_keys) do key
+        read_header(M35070V[1])[key]
+    end
+    # testing rows columns
+    # an additional 3 columns are present because of varibles path, name and hdu
+    @test size(df) == (2, length(relevant_keys) + 3)
+
+    # testing each element in rows and columns
+    @test copy(df[1, :]) == (path = joinpath(dir, "M35070V.fits"), name = "M35070V.fits", hdu = 1, zip(Symbol.(relevant_keys), relevant_values_M35070V)...)
+    @test copy(df[2, :]) == (path = joinpath(dir, "M6707HH.fits"), name = "M6707HH.fits", hdu = 1, zip(Symbol.(relevant_keys), relevant_values_M6707HH)...)
```

Comment:
Style:  might be worth constructing the tuple with line-breaks. instead of one long line